pkg/operation/botanist/kubeapiserver.go

```diff
@@ -153,6 +189,68 @@ func (b *Botanist) computeKubeAPIServerAdmissionPlugins(defaultPlugins, configur
 	return admissionPlugins
 }
 
+func (b *Botanist) ensureAdmissionPluginConfig(plugins []gardencorev1beta1.AdmissionPlugin) ([]gardencorev1beta1.AdmissionPlugin, error) {
+	var (
+		index                     int
+		allowPrivilegedContainers = pointer.BoolDeref(b.Shoot.GetInfo().Spec.Kubernetes.AllowPrivilegedContainers, false)
+		pspDisabled               = b.Shoot.PSPDisabled
+		ok                        bool
+		admissionConfig           = &admissionapiv1beta1.PodSecurityConfiguration{
+			Defaults:   admissionapiv1beta1.PodSecurityDefaults{},
+			Exemptions: admissionapiv1beta1.PodSecurityExemptions{},
+		}
+		userHasSetConfig = false
+	)
+
+	for i, plugin := range plugins {
+		if plugin.Name == "PodSecurity" {
+			index = i
+			break
+		}
+	}
+
+	// if user has set a config in the shoot spec, retrieve it
+	if plugins[index].Config != nil {
+		userHasSetConfig = true
+		config, err := runtime.Decode(codec, plugins[index].Config.Raw)
+		if err != nil {
+			return nil, err
+		}
+		admissionConfig, ok = config.(*admissionapiv1beta1.PodSecurityConfiguration)
+		if !ok {
+			return nil, fmt.Errorf("expected admissionapiv1beta1.PodSecurityConfiguration but got %T", config)
+		}
+	}
+
+	// add kube-system to exempted namespace in all cases
+	if !slices.Contains(admissionConfig.Exemptions.Namespaces, metav1.NamespaceSystem) {
+		admissionConfig.Exemptions.Namespaces = append(admissionConfig.Exemptions.Namespaces, metav1.NamespaceSystem)
+	}
+
+	// if allowPrivilegedContainers is false, overwrite Defaults and enforce restricted level
+	if versionutils.ConstraintK8sGreaterEqual123.Check(b.Shoot.KubernetesVersion) &&
+		pspDisabled &&
+		!allowPrivilegedContainers {
+		admissionConfig.Defaults = admissionapiv1beta1.PodSecurityDefaults{
+			Enforce:        "restricted",
```

Comment:
As discussed, let's rather use `baseline` as default here. Please adapt the docs.